cfgov/core/utils.py

```diff
@@ -24,8 +24,22 @@
 
 LINK_ICON_TEXT_CLASSES = 'a-link_text'
 
-A_TAG = re.compile(r'<a [^>]*?>.+?(?=</a>)</a>(?s)(?i)')
-IMG_TAG = re.compile(r'<img[^>]*>(?i)')
+# Regular expression to match <a> links in HTML strings
+A_TAG = re.compile(
+    # Match an <a containing any attributes
+    r'<a [^>]*?>'
+    # And match everything inside
+    r'.+?'
+    # As long as it's not a </a>, then match '</a>'
+    '(?=</a>)</a>'
+    # Make '.' match new lines, ignore case
+    r'(?s)(?i)'
+)
+
+# If a link contains these elements, it should *not* get an icon
+ICONLESS_LINK_CHILD_ELEMENTS = [
+    'img', 'svg', 'div', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6'
```

Comment:
Possibly we could future-proof this by including [`picture`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture), or wait till which time we actually were to use that element.

Reply:
Actually, this may not be applicable since it contains an `img`, but not sure if the whole thing was wrapped in a link what gets the icon.

Reply:
I'd say disregard and we can add it later if needed.

Reply:
Heh, okay. @Scotchester and I also discussed making this a whitelist instead of a blacklist — which makes more sense to me. It'd be a trivial change based on this, I think, but I want some decision-making and documentation about what kind of links should get the icon and what kind should not before that.